deploy/examples/pool-builtin-mgr.yaml

```diff
@@ -1,16 +1,17 @@
 apiVersion: ceph.rook.io/v1
 kind: CephBlockPool
 metadata:
-  # If the built-in Ceph pool for health metrics needs to be configured with alternate
+  # If the built-in Ceph pool used by the Ceph mgr needs to be configured with alternate
   # settings, create this pool with any of the pool properties. Create this pool immediately
   # with the cluster CR, or else some properties may not be applied when Ceph creates the
   # pool by default.
-  name: device-health-metrics
+  name: builtin-mgr
   namespace: rook-ceph # namespace:cluster
 spec:
   # The required pool name with underscores cannot be specified as a K8s resource name, thus we override
   # the pool name created in Ceph with this name property.
-  name: device_health_metrics
+  # The ".mgr" pool is called "device_health_metrics" in Ceph versions v16.x.y and below.
+  name: .mgr
   failureDomain: host
   replicated:
     size: 3
```

Comment:
Changed this example to show the `.mgr` pool instead of `device_health_metrics` so we could refer to it in the upgrade doc.

Reply:
I hesitated to update this example in the original PR because majority of users will still be on Pacific in rook v1.9. So I wonder if we should create two separate yaml examples, such as keep pool-device-health-metrics.yaml and add pool-builtin-mgr.yaml that is for quincy or newer. But either way it's not a common scenario to create this pool, so maybe it doesn't matter.